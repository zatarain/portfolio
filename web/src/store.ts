import { configureStore, ThunkAction, Action, createSlice } from '@reduxjs/toolkit'

export function makeStore() {
  const responsiveFlipper = createSlice({
    name: 'flipper',
    initialState: {
      responsive: false,
    },
    reducers: {
      flipResponsive: (state) => {
        state.responsive = !state.responsive
      }
    }
  })
  return configureStore({
    reducer: { flipper: responsiveFlipper.reducer },
  })
}

const store = makeStore()

export type AppState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  AppState,
  unknown,
  Action<string>
>

export default store
