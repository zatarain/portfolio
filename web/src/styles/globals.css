@import url('https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css');
@import url('https://fonts.cdnfonts.com/css/cascadia-code');
@import url('react-responsive-carousel/lib/styles/carousel.min.css');
@import url('leaflet/dist/leaflet.css');


:root {
  --bg-page: #fafbfc;
  --bg-primary: #28b873;
  --bg-secondary: #151b28;
  --bg-action: #f40;
  --bg-action-hover: #f60;
  --link-secondary: #a5acba;
  --highlight-secondary: #5cdeac;
  --fg-secondary: #fff;
  --fg-menuitem-hover: #7fffd4;
  --ldn-dark: #29bc9b;
  --ldn-lighter: #aaffec;
  --ldn-light: #79ffe1;
  --ldn: #50e3c2;
  --heading-2: var(--ldn-dark);
  --heading-3: var(--ldn-dark);
  --headings-font: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
}

@media screen {
  html, body {
    padding: 0;
    margin: 0;
    text-align: center;
    font-family: Inter, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu,
      Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
    font-size: 14px;
    scroll-behavior: smooth;
  }

  main {
    padding: 0 3rem;
    margin: 3rem 0;
    font-size: 1.2rem;
    line-height: 1.75rem;
  }
}

* {
  box-sizing: border-box;
}

a:link, a:visited {
  color: var(--ldn-dark);
  text-decoration: none;
}

a:hover, a:hover:visited {
  color: var(--bg-action);
}

.carousel ul.control-dots {
	bottom: 4rem;
	z-index: 3;
}

.carousel ul.control-dots li {
  width: 0.5rem;
  height: 0.5rem;
  padding: 0.5rem;
}

.carousel figcaption {
  margin: 0;
  position: relative;
  bottom: 3rem;
  font-size: 1rem;
}

.leaflet-bottom,
.leaflet-control,
.leaflet-pane,
.leaflet-top {
  z-index: 0 !important;
}

@media screen and (max-width: 850px) {
  .carousel figure img {
    height: 50vh;
  }

  .carousel figcaption {
    bottom: 1rem;
  }

  .carousel ul.control-dots {
    bottom: 3rem;
  }
}

@media screen and (max-width: 539px) {
  .carousel figure img {
    height: 30vh;
  }

  .carousel figcaption {
    bottom: 3rem;
    display: none;
  }

  .carousel ul.control-dots {
    bottom: -1rem;
  }
}
