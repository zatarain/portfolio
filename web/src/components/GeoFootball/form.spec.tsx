import { Provider } from 'react-redux'
import store from '#store'
import { render } from '@testing-library/react'
import { jest } from '@jest/globals'
import MapForm from './form'
import type { GroupedStations, Station } from './types'

jest.mock('react-leaflet', () => {
	return {
		__esModule: true,
		MapContainer: jest.fn(({ children }) => {
			return (<>{children}</>)
		}),
		TileLayer: jest.fn(({ children }) => {
			return (<>{children}</>)
		}),
		Marker: jest.fn(({ children }) => {
			return (<>{children}</>)
		}),
		Popup: jest.fn(({ children }) => {
			return (<>{children}</>)
		}),
		useMapEvents: jest.fn(),
	}
})

jest.mock('react-leaflet-cluster', () => {
	return {
		__esModule: true,
		default: jest.fn(({ children }) => {
			return (<>{children}</>)
		}),
	}
})

jest.mock('next/font/google', () => {
	return {
		__esModule: true,
		Noto_Color_Emoji: jest.fn(() => {
			return {
				className: 'my-font',
			}
		}),
	}
})

describe('<MapForm ...>...</MapForm>', () => {
	it('renders the component correctly', async () => {
		const clusters = {
			GB: [
				{
					id: 1,
					name: 'Canary Wharf',
					slug: 'canary-wharf',
					country: 'GB',
					time_zone: 'Europe/London',
					latitude: 51.50361,
					longitude: -0.01861,
					info_en: 'London Underground Station',
				} as Station,
			],
		} as GroupedStations

		const setClusters = jest.fn()

		const { asFragment } = render(
			<Provider store={store}>
				<MapForm clusters={clusters} setClusters={setClusters} />
			</Provider>
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
