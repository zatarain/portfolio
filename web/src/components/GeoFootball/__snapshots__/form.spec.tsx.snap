// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapForm ...>...</MapForm> renders the component correctly 1`] = `
<DocumentFragment>
  <form
    class="form"
    id="popup-add-form"
    method="POST"
  >
    <h3>
      Add New Marker
    </h3>
    <label
      for="name"
    >
      Name: 
    </label>
    <input
      name="name"
      required=""
      type="text"
    />
    <div
      class="error"
    />
    <div
      class="location"
    >
      <div>
        <label
          for="latitude"
        >
          Latitude: 
        </label>
        <input
          disabled=""
          name="latitude"
          type="text"
          value="51.47800"
        />
      </div>
      <div>
        <label
          for="longitude"
        >
          Longitude: 
        </label>
        <input
          disabled=""
          name="longitude"
          type="text"
          value="0.00000"
        />
      </div>
    </div>
    <div
      class="country-time"
    >
      <div
        class="country"
      >
        <label
          for="country"
        >
          Country: 
        </label>
        <select
          class="my-font"
          name="country"
        >
          <option
            class="my-font"
            value="GB"
          >
            ðŸ‡¬ðŸ‡§
          </option>
        </select>
      </div>
      <div>
        <label
          for="time_zone"
        >
          Timezone: 
        </label>
        <input
          name="time_zone"
          type="text"
        />
      </div>
    </div>
    <label
      for="info_en"
    >
      Additional information: 
    </label>
    <input
      name="info_en"
      type="text"
    />
    <button
      type="submit"
    >
      Save
    </button>
  </form>
</DocumentFragment>
`;
